crypto/mem_dbg.c

```diff
@@ -573,8 +574,8 @@ static void print_leak(const MEM *m, MEM_LEAK *l)
                  m->order, m->file, m->line);
     bufp += strlen(bufp);
 
-    BIO_snprintf(bufp, BUF_REMAIN, "thread=%lu, ",
-                 CRYPTO_THREADID_hash(&m->threadid));
+    tid.tid = m->threadid;
```

Comment:
tid.ldit=0 first?  maybe just cast to unsigned long, or void* and use %p?  just some thoughts, not necesarily recommendations.